test/CMakeLists.txt

```diff
@@ -100,74 +101,91 @@ endfunction()
 
 function(add_test_executable TEST_NAME)
     add_executable(${TEST_NAME} EXCLUDE_FROM_ALL ${ARGN})
-    target_link_libraries(${TEST_NAME} ${CMAKE_THREAD_LIBS_INIT})
-
-    # Cmake does not add flags correctly for gcc
-    if(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
-        set_target_properties(${TEST_NAME} PROPERTIES COMPILE_FLAGS -pthread LINK_FLAGS -pthread)
-    endif()
     set(TEST_COMMAND ${TEST_NAME})
     add_test_command(${TEST_NAME} ${TEST_COMMAND})
     add_dependencies(tests ${TEST_NAME})
     add_dependencies(check ${TEST_NAME})
-    target_link_libraries(${TEST_NAME} migraphx migraphx_onnx migraphx_ref)
-    target_include_directories(${TEST_NAME} PUBLIC include)
-endfunction(add_test_executable)
+    target_compile_features(${TEST_NAME} PRIVATE cxx_std_17)
+    target_link_libraries(${TEST_NAME} PRIVATE Threads::Threads migraphx migraphx_onnx migraphx_ref msgpackc-cxx)
+    target_include_directories(${TEST_NAME} PUBLIC include ${CMAKE_BINARY_DIR}/include)
+endfunction()
 
 file(GLOB TESTS ${CONFIGURE_DEPENDS} *.cpp)
 
 foreach(TEST ${TESTS})
     get_filename_component(BASE_NAME ${TEST} NAME_WE)
     add_test_executable(test_${BASE_NAME} ${TEST})
-    rocm_clang_tidy_check(test_${BASE_NAME})
+    if(CLANG_TIDY_EXE)
+        rocm_clang_tidy_check(test_${BASE_NAME})
+    endif()
 endforeach()
 
 if(MIGRAPHX_ENABLE_GPU)
-    # gpu tests
-    file(GLOB GPU_TESTS ${CONFIGURE_DEPENDS} gpu/*.cpp)
+    set(GPU_TESTS
+        gpu/adjust_allocation.cpp
+        gpu/context_serialize.cpp
+        gpu/hip.cpp
+        gpu/jit.cpp
+        gpu/literal.cpp
+        gpu/manage_host_buffer.cpp
+        gpu/mlir.cpp
+        gpu/pack_args.cpp
+        gpu/pack_int8_args.cpp
+        gpu/quantization.cpp
+        gpu/stream_sync.cpp)
 
     foreach(TEST ${GPU_TESTS})
         get_filename_component(BASE_NAME ${TEST} NAME_WE)
         add_test_executable(test_gpu_${BASE_NAME} ${TEST})
-        rocm_clang_tidy_check(test_gpu_${BASE_NAME})
-        set_tests_properties(test_gpu_${BASE_NAME} PROPERTIES
-            COST 10
-            RESOURCE_LOCK gpu
-        )
+        if(CLANG_TIDY_EXE)
+            rocm_clang_tidy_check(test_gpu_${BASE_NAME})
+        endif()
+        set_tests_properties(test_gpu_${BASE_NAME} PROPERTIES COST 10 RESOURCE_LOCK gpu)
+        target_link_libraries(test_gpu_${BASE_NAME} PRIVATE hip::device MIOpen migraphx_gpu)
```

Comment:
The gpu tests should only link `migraphx_gpu`.  If `MIOpen` is needed to link then the `migraphx_gpu` target was set up wrong. 

Also, `hip::device` is only for gpu-device code which are not in the tests(it should only be linked with `migraphx_device` target as that it the only library which has GPU code).